hugr-model/src/v0/mod.rs

```diff
@@ -662,40 +670,42 @@ pub enum Term<'a> {
     ///
     /// `ctrl : static`
     ControlType,
+
+    /// Constraint that requires a runtime type to be copyable.
+    CopyConstraint {
```

Comment:
I would not object to a general `Term::Constraint { name: str, args: Vec<TermId> }` with the name looked up in *data* that is static for now (i.e. just the one/two constraints and none user-defined). It does kinda say that there will be more constraints coming, but with the data being static there is no possibility of adding them.

Reply:
There is a possibility and moreover a mechanism for adding them: Your `Term::Constraint { name: str, args: Vec<TermId> }` is `Term::Apply`/`Term::ApplyFull` for a constructor that declares a constraint.

There are good arguments for keeping `Term` as minimal as possible by defining anything that does not need special handling as a custom constructor in a standard extension. In particular it'd remove a bunch of boilerplate code. There's also arguments for having anything that is structurally important as builtin; for example `adt`s since the control flow mechanism depends on it, or copyability constraints since they emerge just from the connectivity pattern of ports. I could go either way.